app/components/polaris/resource_item/shortcut_actions_component.html.erb

```diff
@@ -0,0 +1,21 @@
+<%= render(Polaris::BaseComponent.new(**wrapper_arguments)) do %>
+  <%= render(Polaris::ButtonGroupComponent.new(**system_arguments)) do |group| %>
+    <% buttons.each do |button| %>
+      <% group.button(**button.system_arguments) { button.content } %>
+    <% end %>
+  <% end %>
+<% end %>
+
+<% if @persist_actions %>
+  <%= render(Polaris::BaseComponent.new(**disclosure_arguments)) do %>
+    <%= render(Polaris::PopoverComponent.new(alignment: :right)) do |popover| %>
+      <% popover.button(disclosure: :horizontal_dots, plain: true) %>
+
+      <%= render(Polaris::ActionListComponent.new) do |action_list| %>
+        <% buttons.each do |button| %>
+          <% action_list.item(**button.action_list_item_arguments) { button.content } %>
```

Comment:
I like the `ResourceItem::ShortcutActionsComponent` because it can generate a resource item with and without persistent actions. The downside though, as is visible here, we're passing `action_list_item_arguments` from the button to the `action_list.item`.

Which could look something like this when using it:

```erb
<% shortcut_action.button(url: "https://some-url", content: 'Action', action_list_item_arguments: { destructive: true }) %>
```

I'm curious if there is a better way to solve this without making the interface for creating shortcut actions more difficult.